### PR TITLE
Add notice and notice generation scripts

### PR DESCRIPTION
**Pull request by jorchiu**

### CODE REVIEW

File: tools/dotnet_notice_generation.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+cd "$(dirname "$0")/.."
+
+DOTNET_DIRECTORY="cloud_connectors/azure/digital_twins_connector"
+
+dotnet tool install --global dotnet-project-licenses
+mkdir -p "$DOTNET_DIRECTORY/dotnet_licenses_output"
+dotnet-project-licenses -i $DOTNET_DIRECTORY -o -f "$DOTNET_DIRECTORY/dotnet_licenses_output" -u --json -e -c \
+--licenseurl-to-license-mappings "$DOTNET_DIRECTORY/license_url_to_type.json"
+./tools/dotnet_get_licenses.sh "$DOTNET_DIRECTORY/dotnet_licenses_output/licenses.json" "$DOTNET_DIRECTORY/dotnet_licenses_output"
+./tools/dotnet_append_to_notice.sh NOTICE "$DOTNET_DIRECTORY/dotnet_licenses_output/licenses.json"
```
Comment (wilyle) [inline tools/dotnet_notice_generation.sh:12]: the notice file name needs to be configurable like the calling script

File: tools/dotnet_append_to_notice.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
```
Comment (ashbeitz) [inline tools/dotnet_append_to_notice.sh:2]: set -e

see: https://stackoverflow.com/questions/19622198/what-does-set-e-mean-in-a-bash-script

File: tools/dotnet_get_licenses.sh [hunk 3]
```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
```
Comment (ashbeitz) [inline tools/dotnet_get_licenses.sh:2]: set -e

File: tools/dotnet_notice_generation.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
```
Comment (ashbeitz) [inline tools/dotnet_notice_generation.sh:2]: set -e

File: tools/dotnet_notice_generation.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+DOTNET_DIRECTORY="cloud_connectors/azure/digital_twins_connector"
+
+dotnet tool install --global dotnet-project-licenses
+mkdir -p "$DOTNET_DIRECTORY/dotnet_licenses_output"
+dotnet-project-licenses -i $DOTNET_DIRECTORY -o -f "$DOTNET_DIRECTORY/dotnet_licenses_output" -u --json -e -c \
+--licenseurl-to-license-mappings "$DOTNET_DIRECTORY/license_url_to_type.json"
+./tools/dotnet_get_licenses.sh "$DOTNET_DIRECTORY/dotnet_licenses_output/licenses.json" "$DOTNET_DIRECTORY/dotnet_licenses_output"
+./tools/dotnet_append_to_notice.sh "$notice_file_path" "$DOTNET_DIRECTORY/dotnet_licenses_output/licenses.json"
+rm -rf "$DOTNET_DIRECTORY/dotnet_licenses_output"
```
Comment (ashbeitz) [inline tools/dotnet_notice_generation.sh:29]: End the script with an exit 0

File: tools/dotnet_get_licenses.sh [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    # Write updated JSON object to temporary file
+    echo "$updated_json" >> "$temp_file"
+done < <(jq -c '.[]' "$json_file")
+
+# Overwrite original JSON file with updated JSON from temporary file
+jq -s '.' "$temp_file" > "$json_file"
+
+# Remove temporary file
+rm "$temp_file"
```
Comment (ashbeitz) [inline tools/dotnet_get_licenses.sh:55]: End the script with an exit 0

File: tools/dotnet_append_to_notice.sh [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    echo -e "\n#### License\n" >> "$markdown_file"
+    echo '```text' >> "$markdown_file"
+    echo -e "$license_description" >> "$markdown_file"
+    echo '```' >> "$markdown_file"
+done < <(jq -c '.[]' "$json_file")
+
+echo -e "\n## Disclaimer" >> "$markdown_file"
+echo -e "
+This .NET Third Party Licenses list has been generated with [nuget-license](https://github.com/tomchavakis/nuget-license), \
+licensed under [Apache License 2.0](https://github.com/tomchavakis/nuget-license/blob/master/LICENSE)" >> "$markdown_file"
```
Comment (ashbeitz) [inline tools/dotnet_append_to_notice.sh:50]: End the script with an exit 0

**Review by ashbeitz — COMMENTED** [review]
Please take a look at my comments.

File: tools/dotnet_append_to_notice.sh (hunk 2 above)
Comment (jorchiu) [reply tools/dotnet_append_to_notice.sh:2]: Will file an issue to address this later

File: tools/dotnet_get_licenses.sh (hunk 3 above)
Comment (jorchiu) [reply tools/dotnet_get_licenses.sh:2]: Will file an issue to address this later

File: tools/dotnet_notice_generation.sh (hunk 4 above)
Comment (jorchiu) [reply tools/dotnet_notice_generation.sh:2]: Will file an issue to address this later

File: tools/dotnet_notice_generation.sh (hunk 5 above)
Comment (jorchiu) [reply tools/dotnet_notice_generation.sh:29]: Will file an issue to address this later

File: tools/dotnet_get_licenses.sh (hunk 6 above)
Comment (jorchiu) [reply tools/dotnet_get_licenses.sh:55]: Will file an issue to address this later

File: tools/dotnet_append_to_notice.sh (hunk 7 above)
Comment (jorchiu) [reply tools/dotnet_append_to_notice.sh:50]: Will file an issue to address this later

**Review by wilyle — APPROVED** [review]

**Pull request merged**